### PR TITLE
Remove Kafka and ZooKeeper upgrades from StatefulSets

### PR DESCRIPTION
**Pull request by scholzj**
### Type of change

- Task

### Description

Strimzi 0.46 aims to support direct upgrades from Strimzi 0.39 or newer. This is because that is where the KRaft upgrades are supported. So as part of the Strimzi 0.46 release we can remove the logic for upgrading from StatefulSets to StrimziPodSets as the `UseStrimziPodSets` is GA from Strimzi 0.34. This PR removes this logic as well as the related code for managing StatefulSets as we now don't use StatefulSets anymore.

(The full documentation upgrade will be done later as part of the ZooKeeper support removal)

### Checklist

- [x] Make sure all tests pass
- [x] Update documentation
- [x] Check RBAC rights for Kubernetes / OpenShift roles
- [x] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [x] Update CHANGELOG.md

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,7 @@
 
 * [Strimzi EnvVar Configuration Provider](https://github.com/strimzi/kafka-env-var-config-provider) (deprecated in Strimzi 0.38.0) and [Strimzi MirrorMaker 2 Extensions](https://github.com/strimzi/mirror-maker-2-extensions) (deprecated in Strimzi 0.28.0) plugins were removed from Strimzi container images.
   Please use the Apache Kafka [EnvVarConfigProvider](https://github.com/strimzi/kafka-env-var-config-provider?tab=readme-ov-file#deprecation-notice) and [Identity Replication Policy](https://github.com/strimzi/mirror-maker-2-extensions?tab=readme-ov-file#identity-replication-policy) instead.
+* When using Kafka operand and upgrading from Strimzi 0.34 or older, make sure the `UseStrimziPodSets` feature gate is enabled and `StrimziPodSets` are used before upgrading.
```
Comment (PaulRMellor) [inline CHANGELOG.md:9]:
```suggestion
* When using the Kafka operand and upgrading from Strimzi 0.34 or older, make sure the `UseStrimziPodSets` feature gate is enabled and `StrimziPodSet` resources are used before upgrading.
```

File: documentation/modules/operators/ref-operator-cluster-feature-gates.adoc [hunk 2]
```diff
@@ -44,7 +44,9 @@ For example, when you change service account labels or annotations using the `te
 
 The `UseStrimziPodSets` feature gate introduced the `StrimziPodSet` custom resource for managing Kafka and ZooKeeper pods, replacing the use of Kubernetes `StatefulSet` resources.
 
-IMPORTANT: With the `UseStrimziPodSets` feature gate permanently enabled, direct downgrades from Strimzi 0.35 and newer to Strimzi 0.27 or earlier are not possible. You must first downgrade through one of the Strimzi versions in-between, disable the `UseStrimziPodSets` feature gate, and then downgrade to Strimzi 0.27 or earlier.
+IMPORTANT: With the `UseStrimziPodSets` feature gate permanently enabled, direct downgrades from Strimzi 0.35 and newer to Strimzi 0.27 or earlier are not possible.
+You must first downgrade through one of the Strimzi versions in-between, disable the `UseStrimziPodSets` feature gate, and then downgrade to Strimzi 0.27 or earlier.
+When upgrading to Strimzi 0.46.0 or newer from Strimzi 0.34.0 or older while using the Kafka operand, make sure the `UseStrimziPodSets` feature gate is enabled and `StrimziPodSets` are used before starting the upgrade.
```
Comment (PaulRMellor) [inline documentation/modules/operators/ref-operator-cluster-feature-gates.adoc:49]:
```suggestion
Before upgrading from Strimzi 0.38.0 or older to 0.46.0 or newer, make sure the `UseStrimziPodSets` feature gate is enabled and `StrimziPodSet` resources are in use for Kafka.
```

**Review by PaulRMellor — APPROVED** [review]
Thanks. Couple of suggestions

**Review by ppatierno — APPROVED** [review]

**Review by katheris — APPROVED** [review]

**Pull request merged**